modules/cli/src/main/scala/scala/cli/commands/publish/checks/PgpSecretKeyCheck.scala

```diff
@@ -66,228 +88,344 @@ final case class PgpSecretKeyCheck(
-                        "--secret-key",
-                        "publish.secretKey",
-                        configKeys = Seq(Keys.pgpSecretKey.fullName),
-                        extraMessage =
-                          "also specify publish.secretKeyPassword / --secret-key-password if needed." +
-                            (if (options.publishParams.setupCi)
-                               " Alternatively, pass --random-secret-key"
-                             else "")
-                      )
-                    )
-                  }
-              }
+  private lazy val keyServers: Either[BuildException, Seq[Uri]] = {
+    val rawKeyServers = options.sharedPgp.keyServer.filter(_.trim.nonEmpty)
+    if (rawKeyServers.filter(_.trim.nonEmpty).isEmpty)
+      Right(KeyServer.allDefaults)
+    else
+      rawKeyServers
+        .map { keyServerUriStr =>
+          Uri.parse(keyServerUriStr).left.map { err =>
+            new MalformedCliInputError(
+              s"Malformed key server URI '$keyServerUriStr': $err"
+            )
+          }
         }
-      val pushKey: () => Either[BuildException, Unit] = pubKeyOpt match {
+        .sequence
+        .left.map(CompositeBuildException(_))
+  }
+
+  /** Check if the public PGP key is uploaded to all keyservers that were specified
+    */
+  private def isKeyUploaded(pubKeyOpt: Option[PasswordOption]): Either[BuildException, Boolean] =
+    either {
+      pubKeyOpt match {
         case Some(pubKey) =>
           val keyId = value {
             (new PgpProxyMaker).get().keyId(
-              pubKey.value,
+              pubKey.get().value,
               "[generated key]",
               coursierCache,
               logger,
               value(javaCommand),
               options.scalaSigning.cliOptions()
             )
           }
-          val keyServers = value {
-            val rawKeyServers = options.sharedPgp.keyServer.filter(_.trim.nonEmpty)
-            if (rawKeyServers.filter(_.trim.nonEmpty).isEmpty)
-              Right(KeyServer.allDefaults)
-            else
-              rawKeyServers
-                .map { keyServerUriStr =>
-                  Uri.parse(keyServerUriStr).left.map { err =>
-                    new MalformedCliInputError(
-                      s"Malformed key server URI '$keyServerUriStr': $err"
-                    )
-                  }
-                }
-                .sequence
-                .left.map(CompositeBuildException(_))
+
+          value(keyServers).forall { keyServer =>
+            KeyServer.check(keyId, keyServer, backend) match
+              case Right(Right(_)) => true
+              case Right(Left(msg)) =>
+                logger.debug(
+                  s"""Response from $keyServer:
+                     |$msg
+                     |""".stripMargin
+                )
+                false
+              case Left(err) =>
+                logger.debug(s"Error checking $keyId at $keyServer: $err")
```

Comment:
If it is an error message, maybe we should always print it.

```suggestion
                logger.error(s"Error checking $keyId at $keyServer: $err")
```